apiserver/common/instancetypes.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/environs/instances"
+	"github.com/juju/juju/state"
+)
+
+// InstanceTypeBackend provides the methods necessary to obtain provider
+// information.
+type InstanceTypeBackend interface {
+	environs.EnvironConfigGetter
+	ModelTag() names.ModelTag
+	Close() error
+	GetModel(names.ModelTag) (*state.Model, error)
```

Comment:
Don't be so defeatist ;)

Look at ControllerModel in apiserver/cloud/backend.go for an example of how to do this. Anyway, I think this will be moot if you pass the Environ in?